File: src/rust/enum_as_arg.rs
enum Color {
    Red,
    Green,
    Yellow,
    Blue,
}

fn print_color(color: Color) {
    match color {
        Color::Red => println!("red"),
        Color::Green => println!("green"),
        Color::Yellow => println!("yellow"),
    }
}

fn main () {
    let red = Color::Red;
    let _green = Color::Green;
    let _yellow = Color::Yellow;
    print_color(red);
}

// rustc src/rust/enum_as_arg.rs

File: tsEnums/drawbacks/src/typescript/enumAsArg.ts
enum Color {
  Red,
  Green,
  Yellow,
  Blue,
}

function printColor(color: Color) {
  switch (color) {
    case Color.Red:
      console.log('red');
      break;
    case Color.Green:
      console.log('green');
      break;
    case Color.Yellow:
      console.log('yellow');
      break;
  }
}

printColor(0);
